server/shared-models/src/main/scala/com/prisma/shared/models/ProjectJsonFormatter.scala

```diff
@@ -163,7 +165,38 @@ object ProjectJsonFormatter {
     private def addDiscriminator(json: JsObject, fn: Function) = json ++ Json.obj(discriminatorField -> fn.typeCode.toString)
   }
 
-  implicit lazy val relation                  = Json.format[Relation]
+  implicit val relationWrites: Writes[Relation] = (
+    (JsPath \ "name").write[String] and
+      (JsPath \ "description").writeNullable[String] and
+      (JsPath \ "modelAId").write[String] and
+      (JsPath \ "modelBId").write[String] and
+      (JsPath \ "modelAOnDelete").write[OnDelete.Value] and
+      (JsPath \ "modelBOnDelete").write[OnDelete.Value]
+  )(unlift(Relation.unapply))
+
+  implicit val relationReads: Reads[Relation] = {
+    val x = (
+      (JsPath \ "name").read[String] and
+        (JsPath \ "description").readNullable[String] and
+        (JsPath \ "modelAId").read[String] and
+        (JsPath \ "modelBId").read[String] and
+        (JsPath \ "modelAOnDelete").readNullable[OnDelete.Value] and
+        (JsPath \ "modelBOnDelete").readNullable[OnDelete.Value]
+    )
+
+    x.apply { (name, description, modelAId, modelBId, modelAOnDelete, modelBOnDelete) =>
+      Relation(
```

Comment:
While reading the code i realised that we do not need to this explicit invocation of the constructor.
Above we can just do:
```
(JsPath \ "modelAOnDelete").readNullable[OnDelete.Value].map(_.getOrElse(OnDelete.SetNull)) and 
...
```
Therefore we can use the shorter magic syntax again like in:
```
implicit val placeReads: Reads[Place] = (
  (JsPath \ "name").read[String] and
  (JsPath \ "location").read[Location] and
  (JsPath \ "residents").read[Seq[Resident]]
)(Place.apply _)
```

Reply:
Done